src/ape/managers/project.py

```diff
@@ -441,6 +444,69 @@ def _create_source_dict(self, contracts_paths: Collection[Path]) -> Dict[str, So
             for source in contracts_paths
         }
 
+    def run_script(self, name: str, interactive: bool = False):
+        """
+        Run a script from the project ``scripts/`` directory.
+
+        Args:
+            name (str): The script name.
+            interactive (bool): Whether to launch the console as well. Defaults to ``False``.
+        """
+        # Generate the lookup based on all the scripts defined in the project's ``scripts/`` folder
+        # NOTE: If folder does not exist, this will be empty (same as if there are no files)
+        available_scripts = {p.stem: p.resolve() for p in self.scripts_folder.glob("*.py")}
+
+        if Path(name).exists():
+            script_file = Path(name).resolve()
+
+        elif not self.scripts_folder.exists():
+            raise ProjectError("No 'scripts/' directory detected to run script.")
+
+        elif name not in available_scripts:
+            raise ProjectError(f"No script named '{name}' detected in scripts folder.")
+
+        else:
+            script_file = self.scripts_folder / name
+
+        script_path = get_relative_path(script_file, Path.cwd())
+        script_parts = script_path.parts[:-1]
+
+        if any(p == ".." for p in script_parts):
+            raise ProjectError("Cannot execute script from outside current directory")
+
+        # Add to Python path so we can search for the given script to import
+        root_path = Path(".").resolve().root
+        sys.path.append(root_path)
```

Comment:
So now we register the full absolute path.
Then, we import the full path (line 483).

This is how brownie does it and it seems to work 🤷🏻‍♀️